enhancements/insights/prometheus-alerts-for-insights-recommendations.md

```diff
@@ -0,0 +1,268 @@
+      recommendation (do not repeat alerts for previously spotted recommendations)
+    - If a recommendation was previously registered as active and the latest
+      report no longer contains it, the metric will either be reset or
+      unregistered from the Prometheus instance if possible. (remove alerts for
+      no longer hitting recommendations)
+    - If a recommendation was previously registered as inactive or the metric
+      for this recommendation is absent and the latest Insights report contains
+      it, a metric will be updated/created to indicate that the recommendation
+      is currently hitting. (indicate new recommendations)
+4. The metric will have labels containing the ID, severity (based on the total
+   risk property), a human-readable name, and a link to a detailed description
+   of the recommendation and its remediation steps (Insights Advisor URL
+   generated based on the cluster ID and rule ID by IO) and its value will be
+   0/1 to indicate if the recommendation is inactive/active respectively.
+5. There will be a generic/templated info-level alert present on the cluster,
+   which will present the user with brief information about the hitting
+   recommendation (its human-readable name and severity),
+   and the link for details and remediation steps.
+
+It is worth pointing out that if any key part of the Prometheus stack is not
+running as expected on the cluster, then this feature will not work. Prometheus
+manages both the metrics and the alerts, so without it, no alerts can be
+generated by IO.
+
+**Unhealthy Insights Operator scenario:**
+
+- If something goes wrong with IO, we would not want the Insights alerts to
+  remain hanging on the cluster indefinitely.
+- The alert should only fire if IO has been healthy at any point in the past 10
+  minutes. Healthy, in this context, means that there is no major error
+  happening (crashlooping, degraded operator status, etc.). A temporary error
+  state should not cause the alerts to be disabled.
+- If the user intentionally disables IO, the alerts should never fire. If IO is
+  disabled while some of the Insights alerts are firing, the alerts should
+  immediately stop firing (a part of their condition will be that IO is
+  enabled).
+- If IO was disabled by an accident, the user will be notified via the existing
+  alert.
+
+**Unhealthy Smart Proxy scenario:**
+
+- The latest successfully fetched and parsed report is too old
+  (`Meta.LastCheckedAt` is more than 8 hours ago)
+  - Reset the Insights recommendation metrics to avoid showing alerts about
+    recommendations that are no longer hitting.
+- Unable to fetch a report from Smart Proxy (endpoint unreachable, auth error, etc.)
+  - Log error, do not touch the Insights recommendation metrics.
+- The fetched report cannot be decoded into the expected structure, or the data
+  are otherwise invalid
+  - Log error, do not touch the Insights recommendation metrics.
```

Comment:
Not directly related to this proposal but do we already alert on these failure scenarios?

Reply:
Yes we have [InsightsDownloadDegraded](https://github.com/openshift/insights-operator/blob/master/pkg/controller/status/conditions.go#L14) clusteroperator condition and I think it's a good idea to introduce an alert based on this failure.

Reply:
My personal preference would be to stop serving Insights-rule time series immediately.  Having some [time-aggregation in your alert rule][1] would keep the alerts firing for a few minutes, and if a replacement report retrieval had gone through by then, they'd continue firing without interruption.  Otherwise, you'd have the new alert about "we're having trouble pulling Insights reports", and folks would know they were flying blind, and have a link to the hosted Insights dashboard where they could check on things if they were nervous.

[1]: https://github.com/openshift/enhancements/pull/1036#discussion_r818208034